taskvine/src/manager/vine_manager.c

```diff
@@ -2592,6 +2591,25 @@ static vine_result_code_t start_one_task(struct vine_manager *q, struct vine_wor
 		command_line = xxstrdup(t->command_line);
 	}
 
+	/* Send library first if this is a function call and worker doesn't have the library. */
+	int ok = 1;
+	if (t->needs_library && !hash_table_lookup(w->libraries, t->needs_library)) {
+		if (vine_manager_send_library_to_worker(q, w, t->needs_library)) {
+			hash_table_insert(w->libraries,
+					t->needs_library,
+					vine_task_copy(hash_table_lookup(q->libraries, t->needs_library)));
+			debug(D_VINE, "Sending library %s to worker %s\n", t->needs_library, w->workerid);
+		} else {
+			debug(D_VINE, "Cannot send library %s to worker %s\n", t->needs_library, w->workerid);
```

Comment:
This debug could be misleading b/c `vine_manager_send_library_to_worker` may return zero if the library simply does not fit the worker, or is incompatible.  In short, it's not really a failure, it's just the manager realizing (late in the game) that it doesn't make sense to schedule this task here.